modules/peers.js

```diff
@@ -291,20 +312,29 @@ __private.insertSeeds = function(cb) {
 	async.each(
 		library.config.peers.list,
 		(peer, eachCb) => {
+			// Ban peer if it is presented in the array of black listed peers
+			if (__private.isBlacklisted(peer.ip)) {
+				peer.state = Peer.STATE.BANNED;
+			}
+
 			peer = library.logic.peers.create(peer);
 			library.logger.debug(`Processing seed peer: ${peer.string}`);
 			if (library.logic.peers.upsert(peer, true) !== true) {
 				return setImmediate(eachCb);
 			}
-			peer.rpc.status((err, status) => {
-				__private.updatePeerStatus(err, status, peer);
-				if (!err) {
-					updated += 1;
-				} else {
-					library.logger.trace(`Ping peer failed: ${peer.string}`, err);
-				}
-				return setImmediate(eachCb);
-			});
+
+			// Continue if peer it is not blacklisted nor banned
+			if (peer.state != Peer.STATE.BANNED) {
```

Comment:
we should keep the strict checking.
there is no callback called when it's banned